gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveUtils.java

```diff
@@ -101,6 +106,34 @@
   }
 
   /**
+   * For backward compatibility when PathFilter is injected as a parameter.
+   * @param client
+   * @param table
+   * @param filter
+   * @return
+   * @throws IOException
+   */
+  public static List<Partition> getPartitions(IMetaStoreClient client, Table table, Optional<String> filter)
+      throws IOException {
+    return getPartitions(client, table, filter, Optional.<PathFilter>absent());
+  }
+
+  /**
+   * Helping function to determine if the partition has a specific metadata value. Normally used as a filter condition
+   * @param partition
+   * @param expectedMetadataValue
+   * @param checkedMetadataKey
+   *
+   * @return Whether a given partition contains the exact metadata value as expectedMetadataValue.
+   */
```

Comment:
This function is prepared for V2, which inject the time granularity information into partition metadata and filter partitions by checking those metadata.